### PR TITLE
Document the current state of fault tolerance in tests …

### PR DESCRIPTION
**Pull request by JoergSiebahn**
… for server and client implementations using the configured default `ObjectMapper`.

### CODE REVIEW

File: sda-commons-client-jersey/src/test/java/org/sdase/commons/client/jersey/test/ClientTestApp.java [hunk 1]
```diff
@@ -30,6 +31,7 @@ public static void main(String[] args) throws Exception {
   public void initialize(Bootstrap<ClientTestConfig> bootstrap) {
     bootstrap.getObjectMapper().disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
```
Comment (christopher-cudennec) [inline sda-commons-client-jersey/src/test/java/org/sdase/commons/client/jersey/test/ClientTestApp.java:32]: Do you know why the test application uses custom configuration for the `ObjectMapper`?

File: sda-commons-server-jackson/src/test/java/org/sdase/commons/server/jackson/JacksonConfigurationBundleIT.java [hunk 2]
```diff
@@ -597,4 +597,80 @@ public void shouldGetErrorForMissingQueryParameter() {
...
+
+  @Test
+  void shouldNotAcceptSubtypeUndefinedSubtypeThree() {
+    try (Response response =
+        DW.client()
+            .target("http://localhost:" + DW.getLocalPort())
+            .path("subtypes")
+            .request(MediaType.APPLICATION_JSON)
+            .post(Entity.json(Collections.singletonMap("type", "THREE")))) {
+      assertThat(response.getStatus()).isEqualTo(422);
```
Comment (christopher-cudennec) [inline sda-commons-server-jackson/src/test/java/org/sdase/commons/server/jackson/JacksonConfigurationBundleIT.java:635]: That's also my expectation! 👍

File: sda-commons-server-jackson/src/test/java/org/sdase/commons/server/jackson/test/JacksonConfigurationTestApp.java [hunk 3]
```diff
@@ -195,4 +196,22 @@ public Response createRequiredQueryValidationResource(
...
+  @Consumes(MediaType.APPLICATION_JSON)
+  public Response createSubType(@Valid @NotNull AbstractResource requestBody) {
+    return Response.created(URI.create("/subtypes/" + requestBody.getType())).build();
+  }
+
+  @POST
+  @Path("/subtypesTolerant")
+  @Consumes(MediaType.APPLICATION_JSON)
+  public Response createSubTypeAllowNull(@Valid AbstractResource requestBody) {
+    if (requestBody == null) {
```
Comment (christopher-cudennec) [inline sda-commons-server-jackson/src/test/java/org/sdase/commons/server/jackson/test/JacksonConfigurationTestApp.java:211]: I find that a bit speculative to assume that the reasons was that the subtype is not supported. It's unfortunate, that we can't show a proper error message, e.g. `subtype 'three' is not invalid`. We should probably combine `@Valid` and `@NotNull` here, shouldn't we?

File: sda-commons-client-jersey/src/test/java/org/sdase/commons/client/jersey/test/ClientTestApp.java (hunk 1 above)
Comment (JoergSiebahn) [reply sda-commons-client-jersey/src/test/java/org/sdase/commons/client/jersey/test/ClientTestApp.java:32]: That's the OM for reading the config. Imho, we reuse the config for different test apps with different config classes and therefore have properties in the yaml that are not configurable in every test app. But actually, it seems to be the other app that's needs the tolerant reader. Should I try to remove the extra setting in this PR?

Comment (christopher-cudennec) [reply sda-commons-client-jersey/src/test/java/org/sdase/commons/client/jersey/test/ClientTestApp.java:32]: Yeah. Let's see if the extra configuration is needed at all! 👍

**Review by christopher-cudennec — APPROVED** [review]

File: sda-commons-client-jersey/src/test/java/org/sdase/commons/client/jersey/test/ClientTestApp.java (hunk 1 above)
Comment (JoergSiebahn) [reply sda-commons-client-jersey/src/test/java/org/sdase/commons/client/jersey/test/ClientTestApp.java]: Removed: https://github.com/SDA-SE/sda-dropwizard-commons/pull/1680/commits/016003ee131237b4111d4972ca29f90b8f68d795

**Review by christopher-cudennec — APPROVED** [review]

**Pull request merged**